src/world.cpp

```diff
@@ -107,3 +107,19 @@ World::World(
 World::World(Json::Value materials_json, Json::Value biome_data, int tile_type) :
     materials(init_materials(materials_json)),
     terrain_main(3, 3, macro_tile_size, height, 5, tile_type, &materials, biome_data) {}
+
+
+std::vector<entity::Mesh> World::get_mesh_greedy() const {
+        std::vector<entity::Mesh> out;
+        for (const terrain::Chunk& c : terrain_main.get_chunks()) {
+            auto chunk_mesh = entity::generate_mesh(c);
+
+            chunk_mesh.change_color_indexing(materials,
+                terrain::TerrainColorMapping::get_colors_inverse_map());
```

Comment:
I hate this in so many ways. If it's static why does terrain own it.

Reply:
that's where your wrong. its in the terrain namespace. :)

Reply:
Every voxel object will have a ColorMapping. I want to distinguish this one because there will be only in the terrain namespace. It will also be the only statice one.